unit_tests/schema_tests.json

```diff
@@ -0,0 +1,579 @@
+      "pattern_path": "/definitions/PrivateIndividualName/properties/phone_number",
+      "payload_tests": [
+        {
+          "payload": "",
+          "expected": false,
+          "description": "must not be blank"
+        },
+        {
+          "payload": "07234243401",
+          "expected": true,
+          "description": "valid phone numbers start with 07"
+        },
+        {
+          "payload": "082342323401",
+          "expected": false,
+          "description": "does not start with 07 (but right length)"
+        },
+        {
+          "payload": "072342334",
+          "expected": false,
+          "description": "starts with 07 but too short"
+        },
+        {
+          "payload": "082342323143",
+          "expected": false,
+          "description": "starts with 07 but too long"
+        },
+        {
+          "payload": "0723 4243401",
+          "expected": false,
+          "description": "starts with 07 but has a space"
+        },
+        {
+          "payload": "0723 424340",
+          "expected": false,
+          "description": "starts with 07 but has a space"
+        }
+      ]
+    },
+    "address":{
+      "pattern_path": "/definitions/PrivateIndividualName/properties/address",
+      "payload_tests": [
+        {
+          "payload": "",
+          "expected": false,
+          "description": "must not be blank"
+        },
+        {
+          "payload": "PL21 0AQ",
+          "expected": true,
+          "description": "valid post code"
+        },
+        {
+          "payload": "PL21 0CC",
+          "expected": false,
+          "description": "invalid end characters"
+        },
+        {
+          "payload": "PL 0AQ",
+          "expected": false,
+          "description": "invalid start of code"
+        },
+        {
+          "payload": "PL21 0KK",
+          "expected": false,
+          "description": "invalid end characters"
+        },
+        {
+          "payload": "PL210AQ",
+          "expected": false,
+          "description": "must have a space"
+        },
+        {
+          "payload": "PL21 ABC",
+          "expected": false,
+          "description": "Second part must start with a number"
+        }
+      ]
+    },
+    "md_ref":{
+      "pattern_path": "/definitions/md_ref",
+      "payload_tests": [
+        {
+          "payload": "",
+          "expected": false,
+          "description": "must not be blank"
+        },
+        {
+          "payload": "e-MD12345",
+          "expected": true,
+          "description": "valid md ref"
+        },
+        {
+          "payload": "x-XX12345",
+          "expected": false,
+          "description": "must start with e-MD"
+        },
+        {
+          "payload": "e-MD1234",
+          "expected": false,
+          "description": "must have 5 digits"
+        },
+        {
+          "payload": "e-MD12 45",
+          "expected": false,
+          "description": "cannot have spaces"
+        }
```

Comment:
Wondering that it might be worth while introducing a scenario to check the rule that only one alpha character can be on the end if three of four number e-MD reference, so feeding in something like e-MD1234AB and checking that fails validation.